apps/vscode/src/providers/preview/preview-util.ts

```diff
@@ -59,6 +59,30 @@ export function isQuartoShinyKnitrDoc(
   
 }
 
+export async function isRPackage() {
+  const descriptionLines = await parseRPackageDescription();
+	const packageLines = descriptionLines.filter(line => line.startsWith('Package:'));
+	const typeLines = descriptionLines.filter(line => line.startsWith('Type:'));
+	const typeIsPackage = (typeLines.length > 0
+		? typeLines[0].toLowerCase().includes('package')
+		: false);
+	const typeIsPackageOrMissing = typeLines.length === 0 || typeIsPackage;
+	return packageLines.length > 0 && typeIsPackageOrMissing;
+}
+
+async function parseRPackageDescription(): Promise<string[]> {
+	if (vscode.workspace.workspaceFolders !== undefined) {
+		const folderUri = vscode.workspace.workspaceFolders[0].uri;
+		const fileUri = vscode.Uri.joinPath(folderUri, 'DESCRIPTION');
+		try {
+			const bytes = await vscode.workspace.fs.readFile(fileUri);
+			const descriptionText = Buffer.from(bytes).toString('utf8');
+			const descriptionLines = descriptionText.split(/(\r?\n)/);
+			return descriptionLines;
+		} catch { }
```

Comment:
Are we sure we want to eat all exceptions here?

Reply:
I believe so, yes! First, we try to parse the `DESCRIPTION` file. If it works, we return the contents in `descriptionLines` and if anything goes wrong, we return `''` from this function. I'll add an early return with `if(!descriptionLines)` to `isRPackage()` to improve it a bit.

Reply:
Ok! I've been bitten enough by stuff like this that I try to always catch specific exceptions, but we don't have any explicit recommendations (nor are we running eslint here, I don't think), so your explanation is great, thanks!